app/forms/field_handlers/select_handlers.py

```diff
@@ -1,13 +1,60 @@
-from typing import Optional
+from collections import namedtuple
+from typing import Any, Optional, Sequence, Union
 
 from app.forms.field_handlers.field_handler import FieldHandler
 from app.forms.fields import (
     MultipleSelectFieldWithDetailAnswer,
     SelectFieldWithDetailAnswer,
 )
+from app.questionnaire.dynamic_answer_options import DynamicAnswerOptions
 
+Choice = namedtuple("Choice", "value label")
+ChoiceWithDetailAnswer = namedtuple(
+    "ChoiceWithDetailAnswer", "value label detail_answer_id"
+)
+
+ChoiceType = Union[Choice, ChoiceWithDetailAnswer]
+
+
+class SelectHandlerBase(FieldHandler):
+    @property
+    def choices(self) -> Sequence[ChoiceType]:
+        return self._build_dynamic_choices() + self._build_static_choices()
+
+    @property
+    def dynamic_options_schema(self) -> dict[str, Any]:
+        return self.answer_schema.get("dynamic_options", {})
+
+    def _build_dynamic_choices(self) -> list[ChoiceWithDetailAnswer]:
+        if not self.dynamic_options_schema:
+            return []
+
+        dynamic_options = DynamicAnswerOptions(
+            dynamic_options_schema=self.dynamic_options_schema,
+            rule_evaluator=self.rule_evaluator,
+        )
+
+        choices = [
```

Comment:
Would it make it unreadable if we remove `choices` and do immediate return?
```python
return [
            ChoiceWithDetailAnswer(option["value"], option["label"], None)
            for option in dynamic_options.evaluate()
        ]
```

Reply:
I don't think it does, so done 👍